mqtt-simulator/topic.py

```diff
@@ -7,45 +7,55 @@
 from expression_evaluator import ExpressionEvaluator
 
 class Topic(ABC):
-    def __init__(self, broker_url, broker_port, topic_url, topic_data):
+    def __init__(self, broker_url, broker_port, topic_url, topic_data, protocol, clean_session):
+        self.loop = True
         self.broker_url = broker_url
         self.broker_port = broker_port
         self.topic_url = topic_url
         self.topic_data = topic_data
         self.client = None
+        self.protocol = protocol
+        self.clean_session = clean_session
+        self.raw_values_index = 0
```

Comment:
I don't think `raw_values_index` is necessary here. You can move it to `TopicAuto`.